src/ripple/test/jtx/JTx.h

```diff
@@ -62,6 +93,115 @@ struct JTx
     {
         return jv[key];
     }
+
+public:
+    /** Return a property if it exists
+
+        @return nullptr if the Prop does not exist
+    */
+    /** @{ */
+    template <class Prop>
+    Prop*
+    get()
+    {
+        for (auto& prop : props_.list)
+        {
+            if (auto test = dynamic_cast<
+                    prop_type<Prop>*>(
+                        prop.get()))
+                return &test->t;
+        }
+        return nullptr;
+    }
+
+    template <class Prop>
+    Prop const*
+    get() const
+    {
+        for (auto& prop : props_.list)
+        {
+            if (auto test = dynamic_cast<
+                    prop_type<Prop> const*>(
+                        prop.get()))
+                return &test->t;
+        }
+        return nullptr;
+    }
+    /** @} */
+
+    /** Set a property
+        If the property already exists,
+        it is replaced.
+    */
+    /** @{ */
+    void
+    set(std::unique_ptr<basic_prop> p)
+    {
+        for (auto& prop : props_.list)
+        {
+            if (prop->assignable(p.get()))
+            {
+                prop = std::move(p);
+                return;
+            }
+        }
+        props_.list.emplace_back(std::move(p));
+    }
+
+    template <class Prop, class... Args>
+    void
+    set(Args&&... args)
+    {
+        set(std::make_unique<
+            prop_type<Prop>>(
+                std::forward <Args> (
+                    args)...));
+    }
+    /** @} */
+
+private:
+    struct prop_list
```

Comment:
I care a tiny bit less about this since its the unit test interface but if it was in business logic its a tiny bit nicer to keep the main class declaration compact by moving the `prop_list` declaration out. Normally I would forward declare the nested type and put the full definition below the parent class but since the parent class is using this as a data member it can't be an incomplete type.